packages/apps/plugins/plugin-function/src/components/TriggerEditor.tsx

```diff
@@ -19,6 +19,16 @@ import {
 } from '@dxos/functions/types';
 import { Filter, type Space, useQuery } from '@dxos/react-client/echo';
 import { DensityProvider, Input, Select } from '@dxos/react-ui';
+import { distinctBy } from '@dxos/util';
+
+type TriggerId = string;
+
+const stateInitialValues = {
+  schemas: [StackType, SketchType, DocumentType, FileType, MessageType] as any[],
+  selectedSchema: {} as Record<TriggerId, any>,
+};
+
+const state = create<typeof stateInitialValues>(stateInitialValues);
```

Comment:
Lightweight alternative to context. I'm quite fond of this pattern and it's much better for performance to use signals like this.